src/aktualizr_secondary/secondary_tcp_server.cc

```diff
@@ -139,13 +140,20 @@ bool SecondaryTcpServer::HandleOneConnection(int socket) {
         resp->present(AKIpUptaneMes_PR_putMetaResp);
         auto r = resp->putMetaResp();
         r->result = ok ? AKInstallationResult_success : AKInstallationResult_failure;
+        if (ok) {
+          LOG_TRACE << "Image root :\n" << meta_pack.image_root;
+          LOG_TRACE << "Image targets :\n" << meta_pack.image_targets;
+          LOG_TRACE << "Image snapshot :\n" << meta_pack.image_snapshot;
+          LOG_TRACE << "Image timestamp :\n" << meta_pack.image_timestamp;
+        }
```

Comment:
I think these should be at DEBUG level, and they should be printed in the format "Received Image repo Root metadata: ".

I suggest moving these up to right after they are parsed (line 121) and also adding the output statements for the Director metadata.

Reply:
I am not sure if it belongs here, IMHO it should be in AktualizrSecondary context, e.g. AktualizrSecondary::putMetadata.

Reply:
Emm, I think logging here is more like showing what's came from network, logging in AktualizrSecondary is more like showing what's will be put into database. They should be same theoretically. Which is better ? I was a little bit more like the log be close to network, but I'm also open if you think it's better to be close to database.

Reply:
My original suggestion was indeed to print the messages just after decoding them. Currently, we are printing the encoded ASN.1 output, which is pretty useless, and my idea was that we could do a lot better by printing the decoded messages instead.

In fact, for the download and install commands, I'd make a similar suggestion that we should print something as soon as we realize we have a valid message/order from the Primary, and then let the internals of aktualizr-secondary handle printing anything about success or failure.

Reply:
I don't have a strong opinion here, in overall, in the scope of the given design it doesn't really matter. I suppose in an ideal world logging should be in both places, in a message/request dispatcher, and in a message handler.